### PR TITLE
Add documentation example for Perconna installation on Centos

### PR DESCRIPTION
**Pull request by aerostitch**
Hi guys,

Great job on the module.
I was doing the tests to install Percona 5.7 on a Centos 7 box with this module as some people ask if it's doable.
The commit adds this case in the documentation to help the diversity of usage example as this one is pretty easy.

This should help the questions of #757
To have this working, you need to have #818 integrated to the repo.

Thanks for your time,
Joseph

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,6 +167,58 @@ mysql::db { 'mydb':
...
+    },
+    mysqld_safe => {
+      log-error => '/var/log/mysqld.log',
+    },
+  }
+}
+
+class {'mysql::client':
+  package_name => 'Percona-Server-client-57',
+}
```
Comment (solomonty) [inline README.md]: Why is this needed? The client package is a dependency of the server package.

File: README.md [hunk 2]
```diff
@@ -166,6 +167,58 @@ mysql::db { 'mydb':
...
+
+class { 'mysql::bindings':
+  daemon_dev_package_name => 'Percona-Server-devel-57',
```
Comment (solomonty) [inline README.md]: client_dev_package_name

**Comment by solomonty** [comment]
Please take a look at PR #818. I believe it includes some of your changes in addition to other changes needed for Percona.

File: lib/puppet/provider/mysql.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@ class Puppet::Provider::Mysql < Puppet::Provider
   initvars
 
   # Make sure we find mysqld on CentOS
-  ENV['PATH']=ENV['PATH'] + ':/usr/libexec'
+  ENV['PATH']=ENV['PATH'] + ':/usr/libexec:/usr/sbin'
```
Comment (solomonty) [inline lib/puppet/provider/mysql.rb]: I did not need to make this change on CentOS 6

File: lib/puppet/provider/mysql_datadir/mysql.rb [hunk 4]
```diff
@@ -6,7 +6,7 @@
   initvars
 
   # Make sure we find mysqld on CentOS
-  ENV['PATH']=ENV['PATH'] + ':/usr/libexec'
+  ENV['PATH']=ENV['PATH'] + ':/usr/libexec:/usr/sbin'
```
Comment (solomonty) [inline lib/puppet/provider/mysql_datadir/mysql.rb]: I did not need to make this change on CentOS 6

File: README.md (hunk 1 above)
Comment (aerostitch) [inline README.md]: It's not, it's just here to show how to install the client it you want to install it.

File: README.md (hunk 2 above)
Comment (aerostitch) [inline README.md]: Correct, and Percona-Server-shared-57 would be the daemon_dev_package_name. I'll update that.

File: lib/puppet/provider/mysql.rb (hunk 3 above)
Comment (aerostitch) [inline lib/puppet/provider/mysql.rb]: Because it was already in your path. In some minimalist environment it can be absent.

File: lib/puppet/provider/mysql_datadir/mysql.rb (hunk 4 above)
Comment (aerostitch) [inline lib/puppet/provider/mysql_datadir/mysql.rb]: Because it was already in your path. In some minimalist environment it can be absent.

**Comment by aerostitch** [comment]
@solomonty: yeah sorry I didn't see your change before you commented this PR.
I'm updating this to remove the duplicate work, so it will only be about documenting the installation of CentOS on Percona. :)

**Comment by aerostitch** [comment]
Updated, this PR is now just a doc update (but it requires #818 to be integrated in the repo to have this example working) :)

File: README.md [hunk 5]
```diff
@@ -166,6 +167,62 @@ mysql::db { 'mydb':
...
+# also installs Percona-Server-client-57. It is only here to show how to
+# install the Percona MySQL client
+class {'mysql::client':
+  package_name => 'Percona-Server-client-57',
+}
+
+class { 'mysql::bindings':
+  client_dev_package_name => 'Percona-Server-devel-57',
+  daemon_dev_package_name => 'Percona-Server-shared-57',
+}
```
Comment (solomonty) [inline README.md]: It is prudent to include the various ensure options for `mysql::server`, `mysql::client`, and any entries provided in `mysql::bindings` to make sure they are the same version.

File: README.md [hunk 6]
```diff
@@ -166,6 +167,62 @@ mysql::db { 'mydb':
...
+  descr    => 'CentOS $releasever - Percona',
+  baseurl  => 'http://repo.percona.com/centos/$releasever/os/$basearch/',
+  gpgkey   => 'http://www.percona.com/downloads/percona-release/RPM-GPG-KEY-percona',
+  enabled  => 1,
+  gpgcheck => 1,
+}
+
+class {'mysql::server':
+  package_name     => 'Percona-Server-server-57',
+  service_name     => 'mysqld',
```
Comment (solomonty) [inline README.md]: The Percona service name is `mysql` on CentOS 6.

Is it `mysqld` on CentOS 7?

File: README.md [hunk 7]
```diff
@@ -166,6 +167,62 @@ mysql::db { 'mydb':
...
+  config_file      => '/etc/my.cnf',
+  includedir       => '/etc/my.cnf.d',
```
Comment (solomonty) [inline README.md]: Suggest including `root_password` here.

Get it from hiera?

File: README.md (hunk 6 above)
Comment (aerostitch) [inline README.md]: On CentOS 7 both were present.

Comment (aerostitch) [inline README.md]:
```
Service name on CentOS 6 has been renamed from mysqld back to mysql. This change requires manual service restart after being upgraded from Percona Server 5.7.10-1. Bug fixed #1542332.
```
From: https://www.percona.com/doc/percona-server/5.7/release-notes/Percona-Server-5.7.10-2.html

I'll update that. Thanks for the catch! :)

File: README.md (hunk 5 above)
Comment (solomonty) [inline README.md]: The names `client_dev_package_name` and `daemon_dev_package_name` seem confusing. Where do they come from? What does `daemon_dev` mean? I don't see `daemon` in the name of any of the packages.

The Percona MySQL packages are

```
Percona-Server-client-57.x86_64        5.7.11-4.1.el6            percona-release
Percona-Server-devel-57.x86_64         5.7.11-4.1.el6            percona-release
Percona-Server-server-57.x86_64        5.7.11-4.1.el6            percona-release
Percona-Server-shared-57.x86_64        5.7.11-4.1.el6            percona-release
Percona-Server-57-debuginfo.x86_64     5.7.11-4.1.el6            percona-release
Percona-Server-test-57.x86_64          5.7.11-4.1.el6            percona-release
Percona-Server-tokudb-57.x86_64        5.7.11-4.1.el6            percona-release
```

The Oracle MySQL packages are 

```
mysql-community-client.x86_64       5.7.12-1.el6     mysql57-community          
mysql-community-common.x86_64       5.7.12-1.el6     mysql57-community          
mysql-community-devel.x86_64        5.7.12-1.el6     mysql57-community          
mysql-community-embedded.x86_64     5.7.12-1.el6     mysql57-community          
mysql-community-libs.x86_64         5.7.12-1.el6     mysql57-community          
mysql-community-libs-compat.x86_64  5.7.12-1.el6     mysql57-community          
mysql-community-server.x86_64       5.7.12-1.el6     mysql57-community          
mysql-community-test.x86_64         5.7.12-1.el6     mysql57-community          
```

File: README.md [hunk 8]
```diff
@@ -166,6 +167,62 @@ mysql::db { 'mydb':
...
+# Note: this is not needed  in our case as installing Percona-Server-server-57
+# also installs Percona-Server-client-57. It is only here to show how to
+# install the Percona MySQL client
+class {'mysql::client':
+  package_name => 'Percona-Server-client-57',
+}
+
+class { 'mysql::bindings':
+  client_dev_package_name => 'Percona-Server-devel-57',
+  daemon_dev_package_name => 'Percona-Server-shared-57',
```
Comment (solomonty) [inline README.md]: Add these since they default to `false`
```
  client_dev => true,
  daemon_dev => true,
```

in addition to those it may make sense to install the Perl and Python bindings.

```
    perl_enable               => true,
    python_enable             => true,
    perl_package_name         => 'perl-DBD-MySQL',
    python_package_name       => 'MySQL-python',
```

File: README.md (hunk 7 above)
Comment (aerostitch) [inline README.md]: Hahahaha
I was just doing a minimal installation will all defaults, but of course, it totally makes sense! :)
I'll add that.

File: README.md (hunk 8 above)
Comment (aerostitch) [inline README.md]: Lol! :)
You're becoming fancy! ;)
Adding that too. :)

File: README.md (hunk 5 above)
Comment (aerostitch) [inline README.md]: The daemon_dev are generally mysql-devel or mariadb-devel which are the development headers and static libraries.
Those would correspond (big picture) to: Percona-Server-devel-57
https://www.percona.com/doc/percona-server/5.7/installation/yum_repo.html

The client_dev would correspond in this case to mysql-libs (or mysql-community-libs or mysql-commercial-libs ) or mariadb-shared which are shared/dynamic client libraries.
Percona-Server-shared-57 package would correspond to that.
So keeping that logic I should invert those. :)

**Comment by aerostitch** [comment]
Updating the PR taking in account your comments @solomonty.
Thanks! ;)

File: README.md [hunk 9]
```diff
@@ -166,6 +167,78 @@ mysql::db { 'mydb':
...
+  package_ensure => '5.7.11-4.1.el7',
+}
+
+class { 'mysql::bindings':
+  client_dev_package_name   => 'Percona-Server-shared-57',
+  client_dev_package_ensure => '5.7.11-4.1.el7',
+  client_dev                => true,
+  daemon_dev_package_name   => 'Percona-Server-devel-57',
+  daemon_dev_package_ensure => '5.7.11-4.1.el7',
+  daemon_dev                => true,
```
Comment (solomonty) [inline README.md]: Why not add a 
`require => Yumrepo['percona']` 
to each of Class['mysql::server'], Class['mysql::client'], and Class['mysql::bindings']?

File: README.md [hunk 10]
```diff
@@ -166,6 +167,78 @@ mysql::db { 'mydb':
...
+# Dependencies definition
+Yumrepo['percona']->
+Class['mysql::server']
+
+Yumrepo['percona']->
+Class['mysql::client']
+
+Yumrepo['percona']->
+Class['mysql::bindings']
+```
```
Comment (solomonty) [inline README.md]: Why not add a 
`require => Yumrepo['percona']` 
to each of Class['mysql::server'], Class['mysql::client'], and Class['mysql::bindings']?

File: README.md [hunk 11]
```diff
@@ -166,6 +167,78 @@ mysql::db { 'mydb':
...
+# also installs Percona-Server-client-57. It is only here to show how to
+# install the Percona MySQL client
+class {'mysql::client':
+  package_name   => 'Percona-Server-client-57',
+  package_ensure => '5.7.11-4.1.el7',
+}
+
+class { 'mysql::bindings':
+  client_dev_package_name   => 'Percona-Server-shared-57',
+  client_dev_package_ensure => '5.7.11-4.1.el7',
```
Comment (solomonty) [inline README.md]: FYI, when installing `Percona-Server-server-57` it automatically installs `Percona-Server-shared-57`

```

================================================================================
 Package                    Arch     Version            Repository         Size
================================================================================
Installing:
 Percona-Server-server-57   x86_64   5.7.11-4.1.el6     percona-release    39 M
Installing for dependencies:
 Percona-Server-client-57   x86_64   5.7.11-4.1.el6     percona-release   8.2 M
 Percona-Server-shared-57   x86_64   5.7.11-4.1.el6     percona-release   902 k

Transaction Summary
================================================================================
Install       3 Package(s)
```

Comment (aerostitch) [inline README.md]: It's just for the example, I can add a note as for the client if you want. :)

File: README.md (hunk 10 above)
Comment (aerostitch) [inline README.md]: It's just a question of style. I'm more used to do that as you can manage more complex chaining logic.
The result in this specific case would be the same.

**Comment by DavidS** [comment]
Please rebase this on top of the current master, so that travis runs the full-system acceptance tests (new cells in the travis.yml). This will make it much easier to merge tihs.

**Comment by aerostitch** [comment]
@DavidS rebased against master.
